Debug cni

maiqueb opened the pull request:
Adds a very simple meta CNI whose sole purpose is ... to print the result from the previous plugin in the chain.

Useful for educational purposes.

maiqueb committed:
debug-cni: add skeleton

Signed-off-by: Miguel Duarte Barroso <mdbarroso@redhat.com>

maiqueb committed:
debug-cni: add demo script & net config

Signed-off-by: Miguel Duarte Barroso <mdbarroso@redhat.com>

maiqueb committed:
debug-cni, ci: build the binary whenever a PR is pushed

Signed-off-by: Miguel Duarte Barroso <mdbarroso@redhat.com>

danielmellado commented on .github/workflows/check.yml:
```diff
@@ -1,12 +1,12 @@
   push:
     branches:
       - main
-    tags:
```
don't remove this, I left this blank on purpose so it gets triggered on every branch we may want to create.

maiqueb replied:
thinking this through a bit more, I'm unsure the whole `push` makes any sense given how we handle the event (we only lint the bash code & ensure the binary can be built); those actions - IMO -  fit the `pull_request` event,

I wouldn't even define the `push` event until we add some sort of publishing handler (e.g. publish container images for instance).

danielmellado replied:
So, here's my rationale behind this. At some point we may want to be a BDFL and create a push directly. I'd like for that to be linted. Same for tags, not applying now but I was thinking about 'cutting a release' if you know what I mean

maiqueb replied:
so, let's add that once it becomes a necessity. Right now, imo, there's just no reason for it.

danielmellado replied:
once we're done with this let's cut a release and add it

danielmellado commented on cmd/debug-cni.go:
```diff
@@ -0,0 +1,48 @@
+func main() {
+	skel.PluginMain(
+		cmdAdd,
+		nil,
```
I know that you're just piggybacking kubecon-cni, but it may be worth adding an entry for cmdCheck and just put whatever.

maiqueb replied:
I'm OK with it, but, since the purpose of this CNI is to "print" the `prevResult` of chains, the only relevant event is actually `cmdAdd` (the only one with a result).

IMO, I should do the opposite, and undefine `cmdDel`.

danielmellado replied:
That's a good point, but I'd keep them basically for the educational part of that. I'm fine if you drop them off but leave a comment.

maiqueb replied:
makes sense; I'll add a comment explaining the rationale.

danielmellado commented on go.mod:
```diff
@@ -0,0 +1,14 @@
+module github.com/maiqueb/fosdem-2023-cni-chaining
```
we may want to consider vendoring this, just in case there's no network

maiqueb replied:
good point!

maiqueb replied:
Done. Thanks for raising.

danielmellado reviewed (changes requested):
Just some nits, thanks!

maiqueb reviewed (commented):
Thanks for the review.

thoughts @danielmellado ?

maiqueb committed:
build: vendor the project

Vendoring the project is helpful to ship the dependencies along w/ the
source code, which is helpful in disconnected environments.

We need to update the tox configuration since we aren't interested in
linting the vendored scripts.

Signed-off-by: Miguel Duarte Barroso <mdbarroso@redhat.com>
